components/size_aware_image.jsx

```diff
@@ -170,6 +196,79 @@ export default class SizeAwareImage extends React.PureComponent {
             />
         );
 
+        // copyLink, download are two buttons overlayed on image preview
+        // copyLinkTooltip, downloadTooltip are tooltips for the buttons respectively.
+        // if linkCopiedRecently is true, defaultMessage would be 'Copy Link', else 'Copied!'
+
+        const copyLinkTooltip = (
+            <Tooltip
+                id='copy-link-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='single_image_view.copy_link_tooltip'
+                    defaultMessage={this.state.linkCopiedRecently ? 'Copied' : 'Copy link'}
+                />
```

Comment:
We can't change just the `defaultMessage` here when `this.state.linkCopiedRecently` changes. We also need to change the ID to have it pick up a different string from the translation files which are stored separately.

Also, when you do that, you'll need to wrap the IDs in the `t` function from `utils/i18n` so that they're detected by our translation tools. Those tools can pick up an ID automatically from a `FormattedMessage` if it's a constant, but since we're changing the ID, we need to use `t` to do that for us.

After that, it'll look something like
```jsx
<FormattedMessage
    id={this.state.linkCopiedRercently ? t('single_image_view.copy_link_tooltip') : t('single_image_view.copied_link_tooltip')}
    defaultMessage={this.state.linkCopiedRecently ? 'Copied' : 'Copy link'}
/>
```

Reply:
@hmhealey is right about this one, but to improve performance we should probably define 2 different `FormattedMessage` components based on the state:

```
this.state.linkCopiedRercently ? (
    <FormattedMessage
        id={t('single_image_view.copy_link_tooltip')}
        defaultMessage={'Copied'}
    />
) : (
    <FormattedMessage
        id={t('single_image_view.copied_link_tooltip')}
        defaultMessage={'Copy link'}
    />
)
```

Reply:
In that case, you actually don't need the `t` function since the tooling automatically picks up translation IDs from `FormattedMessage`s